nixos/modules/services/networking/prosody.nix

```diff
@@ -1,887 +1,675 @@
-            muc_room_default_public = ${toLua muc.roomDefaultPublic}
-            muc_room_default_members_only = ${toLua muc.roomDefaultMembersOnly}
-            muc_room_default_moderated = ${toLua muc.roomDefaultModerated}
-            muc_room_default_public_jids = ${toLua muc.roomDefaultPublicJids}
-            muc_room_default_change_subject = ${toLua muc.roomDefaultChangeSubject}
-            muc_room_default_history_length = ${toLua muc.roomDefaultHistoryLength}
-            muc_room_default_language = ${toLua muc.roomDefaultLanguage}
-            ${ muc.extraConfig }
-        '') cfg.muc}
-
-      ${ lib.optionalString (cfg.uploadHttp != null) ''
-        -- TODO: think about migrating this to mod-http_file_share instead.
-        Component ${toLua cfg.uploadHttp.domain} "http_upload"
-            http_upload_file_size_limit = ${cfg.uploadHttp.uploadFileSizeLimit}
-            http_upload_expire_after = ${cfg.uploadHttp.uploadExpireAfter}
-            ${lib.optionalString (cfg.uploadHttp.userQuota != null) "http_upload_quota = ${toLua cfg.uploadHttp.userQuota}"}
-            http_upload_path = ${toLua cfg.uploadHttp.httpUploadPath}
-      ''}
-
-      ${ lib.concatStringsSep "\n" (lib.mapAttrsToList (n: v: ''
-        VirtualHost "${v.domain}"
-          enabled = ${boolToString v.enabled};
-          ${ optionalString (v.ssl != null) (createSSLOptsStr v.ssl) }
-          ${ v.extraConfig }
-        '') cfg.virtualHosts) }
-    '';
+            If you know what you're doing, you can disable this warning by
+            setting config.services.prosody.xmppComplianceSuite to false.
+          '';
+        };
+      in
+      [
+        (mkAssertion "muc" ''
+          You need to setup at least a MUC domain to comply with
+          XEP-0423
+        '')
+
+        (mkAssertion "http_upload" ''
+          You need to setup the http_upload module through
+          config.services.prosody.components to comply with
+          XEP-0423.
+        '')
+      ];
 
-    users.users.prosody = mkIf (cfg.user == "prosody") {
-      uid = config.ids.uids.prosody;
-      description = "Prosody user";
-      inherit (cfg) group;
-      home = cfg.dataDir;
+    warnings =
+      optionals (cfg.modules != { }) [ "The option `services.prosody.modules' has been and split into two separate options: `services.prosody.settings.modules_enabled' and `services.prosody.settings.modules_enabled'." ] ++
+      mapAttrsToList (k: v: "The option `services.prosody.virtualHosts.${k}.enabled' has been renamed to `services.prosody.virtualHosts.${k}.settings.enabled'.") (filterAttrs (_: v: v.enabled != null) cfg.virtualHosts) ++
+      mapAttrsToList (k: v: "The option `services.prosody.virtualHosts.${k}.ssl' has been renamed to `services.prosody.virtualHosts.${k}.settings.ssl'.") (filterAttrs (_: v: v.ssl != null) cfg.virtualHosts)
+    ;
+
+    services.prosody.settings = {
+      log = mkDefault { warn = "*syslog"; error = "*syslog"; };
```

Comment:
old default was `log = "*syslog"`; didn't check prosody doc to see if this matters
```diff
-log = "*syslog"
+log = { ["error"] = "*syslog", ["warn"] = "*syslog" }
```

Reply:
Oh right! I didn't want to change any defaults with this PR but that slipped by me! My brain ignored this because I really disagree with the default 😅 

I will fix that up. Thanks!